internal/controller/datadogagent/controller_reconcile_v2.go

```diff
@@ -200,6 +200,14 @@ func (r *Reconciler) reconcileInstanceV2(ctx context.Context, logger logr.Logger
 		errs = append(errs, err)
 		logger.Error(err, "Error cleaning up old DaemonSets")
 	}
+	if err = r.cleanupOldDCADeployments(ctx, logger, instance, resourceManagers, newStatus); err != nil {
```

Comment:
I think we can move all 3 of the `cleanup` methods to have one of the these orders so it's easier to read:
* Have each `cleanup` right after each component's `reconcile`
```
reconcileV2ClusterAgent
cleanupOldDCADeployments

reconcileV2Agent
cleanupExtraneousDaemonSets

reconcileV2ClusterChecksRunner
cleanupOldCCRDeployments
```
* Have all the components' `reconcile`, then have all the components' `cleanup`
```
reconcileV2ClusterAgent
reconcileV2Agent
reconcileV2ClusterChecksRunner

cleanupOldDCADeployments
cleanupExtraneousDaemonSets
cleanupOldCCRDeployments
```